assets/src/dashboard/components/dropdown/index.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useEffect, useMemo, useRef, useState } from 'react';
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as DropDownArrow } from '../../icons/drop-down-arrow.svg';
+import { ReactComponent as DropUpArrow } from '../../icons/drop-up-arrow.svg';
+
+import PopoverMenu from '../popover-menu';
+
+const DropdownContainer = styled.div`
+  position: relative;
+`;
+
+const Label = styled.label`
+  display: flex;
+  flex-direction: column;
+`;
+
+export const InnerDropdown = styled.button`
+  align-items: center;
+  background-color: ${({ theme }) => theme.colors.gray25};
+  border-radius: 4px;
+  border: 1px solid transparent;
+  color: ${({ theme }) => theme.colors.gray600};
+  cursor: ${({ disabled }) => (disabled ? 'inherit' : 'pointer')};
+  display: flex;
+  font-family: ${({ theme }) => theme.fonts.dropdown.family};
+  font-size: ${({ theme }) => theme.fonts.dropdown.size};
+  font-weight: ${({ theme }) => theme.fonts.dropdown.weight};
+  height: 40px;
+  justify-content: space-between;
+  letter-spacing: ${({ theme }) => theme.fonts.dropdown.letterSpacing};
+  line-height: ${({ theme }) => theme.fonts.dropdown.lineHeight};
+  padding: 10px 16px;
+  width: 100%;
+
+  &:disabled {
+    color: ${({ theme }) => theme.colors.gray400};
+  }
+`;
+InnerDropdown.propTypes = {
+  disabled: PropTypes.bool,
+  isOpen: PropTypes.bool,
+};
+
+const InnerDropdownText = styled.span`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+const DropdownIcon = styled.span`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  & > svg {
+    color: ${({ theme }) => theme.colors.gray300};
+    width: 10px;
+    height: 5px;
+  }
+`;
+
+const Dropdown = ({
+  ariaLabel,
+  items,
+  disabled,
+  onChange,
+  value,
+  placeholder,
+  ...rest
+}) => {
+  const [showMenu, setShowMenu] = useState(false);
+  const dropdownRef = useRef();
+
+  const handleOutsideClick = (e) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(e.target) &&
+      showMenu === true
+    ) {
+      setShowMenu(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleOutsideClick);
```

Comment:
This needs to be a capture listener.

Reply:
In the editor we have `useFocusOut`, which deals with both outside click and focus. Can we reuse that?

Reply:
We can! For `utils` I know Paul doesn't want us importing outside of `edit-story` or `dashboard`. We can copy this file into the `dashboard` dir and then create a ticket to unify common utils down the road.

Reply:
Sounds good 👍

Reply:
oh awesome, thanks everyone for making this update so easy! Added the util  - works great